scim2-sdk-common/src/main/java/com/unboundid/scim2/common/utils/Parser.java

```diff
@@ -122,12 +123,23 @@ public void reset()
 
     /**
      * {@inheritDoc}
+     *
+     * @throws IllegalArgumentException   If {@code n} is negative.
      */
     @Override
-    public long skip(final long n)
+    public long skip(final long n) throws IllegalArgumentException
     {
+      if (n < 0L)
+      {
+        throw new IllegalArgumentException(
+            "Attempted to skip a negative number of characters.");
+      }
```

Comment:
I needed to ensure that the input value was not a very large negative number. The parent class throws an IllegalArgumentException for any negative value, so it made sense to include that here too.